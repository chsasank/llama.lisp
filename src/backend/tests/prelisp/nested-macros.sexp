(c-lisp
  (define ((__add_dot4x4 void)
           (k int)
           (a (ptr float))
           (lda int)
           (b (ptr float))
           (ldb int)
           (c (ptr float))
           (ldc int))
    (store (ptradd c (add 0 (mul 0 ldc))) 0)
    (for ((set p 0) (lt p k) (set p (add p 1)))
        ,(muladd
            ,(arr-idx c ldc 0 0)
            ,(arr-idx a ldc 0 p)
            ,(arr-idx b ldc p 0))
    (ret))))
