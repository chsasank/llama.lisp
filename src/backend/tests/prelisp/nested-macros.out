(c-lisp
  (define ((__add_dot4x4 void)
           (k int)
           (a (ptr float))
           (lda int)
           (b (ptr float))
           (ldb int)
           (c (ptr float))
           (ldc int))
    (store (ptradd c (add 0 (mul 0 ldc))) 0)
    (for ((set p 0) (lt p k) (set p (add p 1)))
         (store (ptradd c (add 0 (mul 0 ldc)))
                (fadd (load (ptradd c (add 0 (mul 0 ldc))))
                      (fmul (load (ptradd a (add 0 (mul p ldc))))
                            (load (ptradd b (add p (mul 0 ldc)))))))
         (ret))))
